crates/log_probe/src/lib.rs

```diff
@@ -0,0 +1,128 @@
+use composer_api::{self, Client, LogStats};
+use log::{self, Level};
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        mpsc::{sync_channel, Receiver, SyncSender},
+        Arc,
+    },
+    time::{Duration, Instant},
+};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum LogProbeError {
+    #[error("Network error: {0}")]
+    NetworkError(String),
+}
+
+pub struct LogProbe {
+    tx: SyncSender<AggregatorMessage>,
+    shutdown: Arc<AtomicBool>,
+}
+
+impl Drop for LogProbe {
+    fn drop(&mut self) {
+        self.shutdown.store(true, Ordering::Relaxed);
+    }
+}
+
+enum AggregatorMessage {
+    AddRecord(Level),
+    Tick,
+}
+
+impl LogProbe {
+    pub fn new(
+        server_address: Option<String>,
+        report_frequency: Duration,
+    ) -> Result<Self, LogProbeError> {
+        let shutdown = Arc::<AtomicBool>::default();
+        let client = if let Some(address) = server_address {
+            Client::new(address)
+        } else {
+            Client::try_default()
+        }
+        .map_err(|e| LogProbeError::NetworkError(format!("{e}")))?;
+
+        let (tx, rx) = sync_channel::<AggregatorMessage>(100);
+        spawn_aggregator_thread(rx, client, shutdown.clone());
+        spawn_tick_thread(tx.clone(), shutdown.clone(), report_frequency);
+
+        Ok(Self { tx, shutdown })
+    }
+}
+
+fn spawn_tick_thread(
+    tx: SyncSender<AggregatorMessage>,
+    shutdown: Arc<AtomicBool>,
+    report_frequency: Duration,
```

Comment:
nit: since the type is a `Duration` a better fitting name could be `report_period`.